wgpu-native/src/instance.rs

```diff
@@ -168,11 +175,12 @@ pub extern "C" fn wgpu_instance_create_surface_from_windows_hwnd(
     HUB.surfaces.register_local(surface)
 }
 
-pub fn instance_get_adapter(instance_id: InstanceId, desc: &AdapterDescriptor) -> AdapterHandle {
-    let instance_guard = HUB.instances.read();
-    let instance = &instance_guard[instance_id];
+fn get_adapter(
+    adapters: Vec<hal::Adapter<back::Backend>>,
```

Comment:
let's turn this into a slice: `&[hal::Adapter<B>]`

Reply:
It returns an adapter by value not by reference, so it can't take the possible set of adapters by reference.  I can change this to `I: IntoIterator<Item = hal::Adapter<B>>`?

Reply:
Actually it won't matter anymore since the methods will be combined again.